src/App.tsx

```diff
@@ -1,28 +1,75 @@
-import { useState } from "react";
-import { getPublicKey } from "./tauriCommands";
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "./components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "./components/ui/dialog";
+import { handleSignEventRequests } from "./tauriCommands";
 
 const App = () => {
   const [publicKey, setPublicKey] = useState("");
+  const [open, setOpen] = useState(false);
+  const [event, setEvent] = useState({});
+  const resolveRejectRef = useRef<{
+    resolve: (value: boolean) => void;
+    reject: (value: boolean) => void;
+  } | null>(null);
+
+  useEffect(() => {
+    const handleEvent = (event: any): Promise<boolean> => {
```

Comment:
Can we use the existing `UnsignedNostrEvent` interface instead of `any`? Or if you inline this function into the call to `handleSignEventRequests()`, you don't have to define the type at all. Your call!